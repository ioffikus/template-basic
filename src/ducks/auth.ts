import { createAction, createReducer } from 'redux-act';
import { AxiosResponse } from 'axios';
import { push } from 'connected-react-router';
import { GET, POST } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'AUTH';
const NS = `${REDUCER}__`;

export interface IAuthState extends Store.IReduxCommonState {
  isAuthenticated: boolean;
  isAuthLinkRequesting: boolean;
  url: string;
  data: object;
}

export const initialState: IAuthState = {
  isAuthenticated: false,
  isRequesting: false,
  isRequestError: false,
  isAuthLinkRequesting: false,
  url: null,
  data: null,
};

const reducer = createReducer<IAuthState>({}, initialState);

export const loginUserSuccess = createAction(`${NS}LOGIN_USER_SUCCESS`);
reducer.on(loginUserSuccess, state => ({
  ...state,
  isRequesting: false,
  isRequestError: false,
  isAuthenticated: true,
}));

const authLinkSuccess = createAction<{ url: string; data: object }>(`${NS}AUTH_LINK_SUCCESS`);
reducer.on(authLinkSuccess, (state, { url, data }) => ({
  ...state,
  url,
  data: { ...data },
  isAuthLinkRequesting: false,
}));

const loginUserFailure = createAction(`${NS}LOGIN_USER_FAILURE`);
reducer.on(loginUserFailure, state => ({
  ...state,
  isRequesting: false,
  isAuthenticated: false,
  isRequestError: true,
}));

const authLinkFailure = createAction(`${NS}AUTH_LINK_FAILURE`);
reducer.on(authLinkFailure, state => ({
  ...state,
  isAuthLinkRequesting: false,
  url: null,
  data: null,
}));

const loginUserRequest = createAction(`${NS}LOGIN_USER_REQUEST`);
reducer.on(loginUserRequest, state => ({
  ...state,
  isRequesting: true,
}));

const authLinkRequest = createAction(`${NS}AUTH_LINK_REQUEST`);
reducer.on(authLinkRequest, state => ({
  ...state,
  isAuthLinkRequesting: true,
  url: null,
  data: null,
}));

const reset = createAction(`${NS}RESET`);
reducer.on(reset, () => ({
  ...initialState,
}));

export const getAuthLink = (): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(authLinkRequest());
    GET(
      {
        url: `${EndpointsConfig.AUTH.AUTH_LINK}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        const { url, data } = resp.data as { url: string; data: object };
        // WIP does not use Alicanto.API.GET_AUTH_OAUTH_PROVIDER_URL_RESPONSE for develop
        dispatch(authLinkSuccess({ url, data }));
        resolve();
      })
      .catch(error => {
        dispatch(authLinkFailure());
        reject(error);
      });
  });
};

export const loginUser = (username: string, password: string): Store.IReduxThunkCb => (
  dispatch,
  getState,
  xShopHeaders,
) => {
  return new Promise((resolve, reject) => {
    dispatch(loginUserRequest());
    const data: Alicanto.API.POST_AUTH_LOGIN_BODY = {
      username,
      password,
    };
    POST(
      {
        data,
        url: `${EndpointsConfig.AUTH.LOGIN}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(loginUserSuccess());
        if (process.env.BROWSER) {
          // TODO: move to external
          // redirect to home page
          dispatch(push('/'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(loginUserFailure());
        reject(error);
      });
  });
};

export const logoutUser = (cb: () => void): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(loginUserRequest());
    GET(
      {
        url: `${EndpointsConfig.AUTH.LOGOUT}`,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(loginUserSuccess());
        dispatch(reset());
        if (cb) {
          cb();
        }
        if (process.env.BROWSER) {
          // TODO: move to external
          // redirect to home page
          dispatch(push('/'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(loginUserFailure());
        reject(error);
      });
  });
};

export default reducer;
