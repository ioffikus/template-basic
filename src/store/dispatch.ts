import { Store } from 'redux';
import _ from 'lodash';
import { showError } from 'src/store/common';

let store: Store<Store.IState>;

export const setupDispatch = (config: { store: Store<Store.IState> }) => {
  store = config.store;
};

const dispatch = (action: any) => {
  if (_.isFunction(action)) {
    return store.dispatch(action).catch((error: any) => {
      if (!_.isNull(error)) {
        showError(error);
      }
      return Promise.reject(error);
    });
  }
  return store.dispatch(action);
};

export default dispatch;
