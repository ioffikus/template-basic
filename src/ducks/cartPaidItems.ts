import { createAction, createReducer } from 'redux-act';
import { AxiosResponse } from 'axios';
import { ICardPaidItemsState } from 'src/ducks/cartPaidItems';
import _ from 'lodash';
import { GET } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'CART_PAID_ITEMS';
const NS = `${REDUCER}__`;

export interface ICardPaidItemsState extends Alicanto.API.GET_CART_PAID_ITEMS_RESPONSE, Store.IReduxCommonState {}

export const initialState: ICardPaidItemsState = {
  next: null,
  previous: null,
  count: 0,
  results: [],
  isRequestError: false,
  isRequesting: false,
};

const reducer = createReducer<ICardPaidItemsState>({}, initialState);

const callFailure = createAction(`${NS}CALL_FAILURE`);
reducer.on(callFailure, state => ({
  ...state,
  next: null,
  previous: null,
  count: 0,
  results: [],
  isRequestError: true,
  isRequesting: false,
}));

const callSuccess = createAction<Alicanto.API.GET_CART_PAID_ITEMS_RESPONSE>(`${NS}CALL_SUCCESS`);
reducer.on(callSuccess, (state, { next, previous, count, results }) => ({
  ...state,
  next,
  previous,
  count,
  results: [...results],
  isRequestError: false,
  isRequesting: false,
}));

const callRequest = createAction(`${NS}CALL_REQUEST`);
reducer.on(callRequest, state => ({
  ...state,
  isRequesting: true,
}));

interface ICartPaidItemsQuery {
  page?: string;
  page_size?: string;
}

const getCartPaidItemsRequestParams = ({ page, page_size }: ICartPaidItemsQuery) => {
  const params: ICartPaidItemsQuery = {
    page_size: '10',
  };
  if (!_.isNil(page)) {
    params.page = page;
  }
  if (!_.isNil(page_size)) {
    params.page_size = page_size;
  }
  return params;
};

export const getCartPaidItems = (page = '1'): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(callRequest());
    GET(
      {
        url: `${EndpointsConfig.CART_PAID_ITEMS}`,
        params: getCartPaidItemsRequestParams({ page }),
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        const data = resp.data as Alicanto.API.GET_CART_PAID_ITEMS_RESPONSE;
        let results = data.results;
        if (page !== '1') {
          const oldResults = getState().cartPaidItems.results;
          results = [...oldResults, ...results];
        }
        dispatch(callSuccess({ ...data, results }));
        resolve();
      })
      .catch(error => {
        dispatch(callFailure());
        reject(error);
      });
  });
};

export default reducer;
