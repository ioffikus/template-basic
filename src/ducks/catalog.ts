import { createAction, createReducer } from 'redux-act';
import { AxiosResponse } from 'axios';
import _ from 'lodash';
import { ICatalogQueryState } from 'src/ducks/catalogQuery';
import { GET } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'CATALOG';
const NS = `${REDUCER}__`;

export interface ICatalogState extends Alicanto.API.GET_CATALOG_RESPONSE, Store.IReduxCommonState {}

export const initialState: ICatalogState = {
  next: null,
  previous: null,
  count: 0,
  results: [],
  isRequesting: false,
  isRequestError: false,
};

const reducer = createReducer<ICatalogState>({}, initialState);

const callSuccess = createAction<Alicanto.API.GET_CATALOG_RESPONSE>(`${NS}CALL_SUCCESS`);
reducer.on(callSuccess, (state, { next, previous, count, results }) => ({
  ...state,
  next,
  previous,
  count,
  results: [...results],
  isRequesting: false,
  isRequestError: false,
}));

const callFailure = createAction(`${NS}CALL_FAILURE`);
reducer.on(callFailure, state => ({
  ...state,
  next: null,
  previous: null,
  count: 0,
  results: [],
  isRequesting: false,
  isRequestError: true,
}));

const callRequest = createAction(`${NS}CALL_REQUEST`);
reducer.on(callRequest, state => ({
  ...state,
  isRequesting: true,
}));

interface ICatalogQuery {
  search?: string;
  page?: string;
  page_size: string;
  ordering?: string;
  filter?: string;
}

const getCatalogRequestParams = ({ search, page, pageSize, ordering, filterObj }: ICatalogQueryState) => {
  const params: ICatalogQuery = {
    page_size: pageSize,
  };
  if (!_.isEmpty(search)) {
    params.search = search;
  }
  if (!_.isEmpty(page)) {
    params.page = page;
  }
  if (!_.isEmpty(ordering)) {
    params.ordering = ordering;
  }
  if (!_.isEmpty(filterObj)) {
    params.filter = JSON.stringify(filterObj);
  }
  return params;
};

export const getCatalog = (abortToken: string): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(callRequest());
    const catalogQuery = getState().catalogQuery;
    GET(
      {
        url: EndpointsConfig.CATALOG,
        params: getCatalogRequestParams(catalogQuery),
      },
      xShopHeaders,
      abortToken,
    )
      .then((resp: AxiosResponse) => {
        const data = resp.data as Alicanto.API.GET_CATALOG_RESPONSE;
        let results = data.results;
        if (catalogQuery.page !== '1') {
          const oldResults = getState().catalog.results;
          results = [...oldResults, ...results];
        }
        dispatch(callSuccess({ ...data, results }));
        resolve();
      })
      .catch(error => {
        dispatch(callFailure());
        reject(error);
      });
  });
};

export default reducer;
