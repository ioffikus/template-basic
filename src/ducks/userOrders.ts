import { createAction, createReducer } from 'redux-act';
import _ from 'lodash';
import { BusinessLogicError } from 'src/store/common';
import { deleteCart } from 'src/ducks/cart';
import { AxiosResponse } from 'axios';
import { push } from 'connected-react-router';
import { GET, POST } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'USER_ORDERS';
const NS = `${REDUCER}__`;

export interface IUserOrdersState extends Alicanto.API.GET_USER_ORDERS_RESPONSE, Store.IReduxCommonState {
  currentPaymentId: number;
}

export const initialState: IUserOrdersState = {
  count: 0,
  next: null,
  previous: null,
  results: [],
  currentPaymentId: null,
  isRequestError: false,
  isRequesting: false,
};

const reducer = createReducer<IUserOrdersState>({}, initialState);

const getFailure = createAction(`${NS}GET_FAILURE`);
reducer.on(getFailure, state => ({
  ...state,
  isRequestError: true,
  isRequesting: false,
}));

const getSuccess = createAction<Alicanto.API.GET_USER_ORDERS_RESPONSE>(`${NS}GET_SUCCESS`);
reducer.on(getSuccess, (state, { next, previous, count, results }) => ({
  ...state,
  next,
  previous,
  count,
  results: [...results],
  isRequestError: false,
  isRequesting: false,
}));

const getRequest = createAction(`${NS}GET_REQUEST`);
reducer.on(getRequest, state => ({
  ...state,
  isRequestError: false,
  isRequesting: true,
}));

const postFailure = createAction(`${NS}POST_FAILURE`);
reducer.on(postFailure, state => ({
  ...state,
  count: 0,
  next: null,
  previous: null,
  results: [],
  currentPaymentId: null,
  isRequestError: true,
  isRequesting: false,
}));

const postSuccess = createAction(`${NS}POST_SUCCESS`);
reducer.on(postSuccess, state => ({
  ...state,
  isRequestError: false,
  isRequesting: false,
}));

const postRequest = createAction(`${NS}POST_REQUEST`);
reducer.on(postRequest, state => ({
  ...state,
  isRequestError: false,
  isRequesting: true,
}));

const setCurrentPaymentId = createAction<number>(`${NS}SET_CURRENT_PAYMENT_ID`);
reducer.on(setCurrentPaymentId, (state, currentPaymentId) => ({
  ...state,
  currentPaymentId,
}));

interface IUserOrdersQuery {
  page?: string;
  page_size?: string;
}

const getUserOrdersRequestParams = ({ page, page_size }: IUserOrdersQuery) => {
  const params: IUserOrdersQuery = {
    page_size: '3',
  };
  if (!_.isNil(page)) {
    params.page = page;
  }
  if (!_.isNil(page_size)) {
    params.page_size = page_size;
  }
  return params;
};

export const getUserOrders = (page = '1'): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(getRequest());
    GET(
      {
        url: `${EndpointsConfig.USER_ORDERS}`,
        params: getUserOrdersRequestParams({ page }),
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        const data = resp.data as Alicanto.API.GET_USER_ORDERS_RESPONSE;
        let results = data.results;
        results = results.map((item: Alicanto.Models.Order) => ({
          ...item,
          cart: {
            ...item.cart,
            total: parseFloat(_.get(item.cart, 'total', '0')),
          },
        }));
        if (page !== '1') {
          const oldResults = getState().userOrders.results;
          results = [...oldResults, ...results];
        }
        dispatch(getSuccess({ ...data, results }));
        resolve();
      })
      .catch(error => {
        dispatch(getFailure());
        reject(error);
      });
  }).then(() => {
    const currentPaymentId = getState().userOrders.currentPaymentId;
    if (process.env.BROWSER && _.isNull(currentPaymentId)) {
      const lastPaymentTimeInMs = window.localStorage.getItem('paymentTime');
      const lastPaymentId = window.localStorage.getItem('paymentId');
      if (lastPaymentTimeInMs && lastPaymentId) {
        const msNow = Date.now();
        const passingTime = msNow - parseInt(lastPaymentTimeInMs, 10);
        if (passingTime < 30 * 1000) {
          dispatch(setCurrentPaymentId(parseInt(lastPaymentId, 10)));
          setTimeout(() => {
            dispatch(setCurrentPaymentId(null));
            window.localStorage.removeItem('paymentTime');
            window.localStorage.removeItem('paymentId');
          }, 30 * 1000 - passingTime);
        }
      }
    }
  });
};

export const checkoutUserOrder = (): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    const isAuthenticated = getState().auth.isAuthenticated;
    if (!isAuthenticated) {
      const anonCheckout = getState().ui.routes.cart.errors.anonCheckout.i18n.defaultText;
      const error = new BusinessLogicError(anonCheckout);
      reject(error);
      return;
    }
    dispatch(postRequest());
    const data: Alicanto.API.POST_USER_ORDERS_BODY = { cart: getState().cart.id };
    POST(
      {
        data,
        url: EndpointsConfig.USER_ORDERS,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(deleteCart());
        dispatch(postSuccess());
        if (process.env.BROWSER) {
          //  TODO: move to external?
          dispatch(push('/user/orders'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(postFailure());
        reject(error);
      });
  });
};

export const payAnOrder = (id: number): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    const currentPaymentId = getState().userOrders.currentPaymentId;
    if (currentPaymentId) {
      const ui = getState().ui;
      reject(new BusinessLogicError(ui.routes.user.orders.paymentWarning.i18n.defaultText));
      return;
    }
    if (process.env.BROWSER) {
      window.localStorage.setItem('paymentTime', '' + Date.now());
      window.localStorage.setItem('paymentId', `${id}`);
    }
    dispatch(setCurrentPaymentId(id));
    setTimeout(() => {
      dispatch(setCurrentPaymentId(null));
      if (process.env.BROWSER) {
        window.localStorage.removeItem('paymentTime');
        window.localStorage.removeItem('paymentId');
      }
    }, 30 * 1000);
    dispatch(postRequest());
    const data: Alicanto.API.POST_PAYMENTS_PAY_BODY = { order: id + '' };
    POST(
      {
        data,
        url: EndpointsConfig.PAY_ORDER,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(postSuccess());
        const data = resp.data as Alicanto.API.POST_PAYMENTS_PAY_RESPONSE;
        const redirect_url = data.redirect_to.redirect_url;
        if (process.env.BROWSER) {
          window.location.assign(redirect_url);
        }
        resolve();
      })
      .catch(error => {
        dispatch(postFailure());
        reject(error);
      });
  });
};

export default reducer;
