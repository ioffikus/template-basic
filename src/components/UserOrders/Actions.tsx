import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'antd';
import dispatch from 'src/store/dispatch';
import { payAnOrder } from 'src/ducks/userOrders';

interface IOwnProps {
  isPaid: boolean;
  id: number;
}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  currentPaymentId: number;
}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState) => ({
  ui: state.ui,
  currentPaymentId: state.userOrders.currentPaymentId,
});

const WrappedActions = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Actions extends React.Component<IProps, IState> {
    handleClick = () => {
      const { id } = this.props;
      dispatch(payAnOrder(id));
    };

    render() {
      const { isPaid, id, currentPaymentId, ui } = this.props;
      const buttonUI = ui.routes.user.orders.button.i18n.defaultText;
      return (
        <div className="root">
          {!isPaid && (
            <Button
              type="primary"
              size="small"
              onClick={this.handleClick}
              key="button"
              loading={currentPaymentId === id}
            >
              {buttonUI}
            </Button>
          )}
        </div>
      );
    }
  },
);

export default WrappedActions;
