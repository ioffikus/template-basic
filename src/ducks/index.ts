// instead this, we have router in store via connected-react-router
// import { routerReducer as routing, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';
import auth from './auth';
import cart from './cart';
import cartPaidItems from './cartPaidItems';
import catalogQuery from './catalogQuery';
import catalog from './catalog';
import categories from './categories';
import categoriesMenu from './categoriesMenu';
import filter from './filter';
import locale from './locale';
import meta from './meta';
import page from './page';
import product from './product';
import progress from './progress';
import popularItems from './popularItems';
import recommendItems from './recommendItems';
import routes from './routes';
import ui from './ui';
import user from './user';
import userDelivery from './userDelivery';
import userOrders from './userOrders';

export default combineReducers<Store.IState>({
  ui,
  auth,
  user,
  categories,
  categoriesMenu,
  catalogQuery,
  catalog,
  product,
  progress,
  routes,
  meta,
  filter,
  cart,
  cartPaidItems,
  userDelivery,
  userOrders,
  popularItems,
  recommendItems,
  page,
  locale,
});
