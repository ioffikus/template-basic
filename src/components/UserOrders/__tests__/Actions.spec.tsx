import React from 'react';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Button } from 'antd';
import { setupDispatch } from 'src/store/dispatch';
import uiStateMock from 'src/store/__mocks__/uiStore';
import userOrdersStateMock, { initialState as userOrdersInitialState } from 'src/store/__mocks__/userOrdersStore';
import { payAnOrder } from 'src/ducks/userOrders';
import Actions from '../Actions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('src/ducks/userOrders', () => {
  return {
    payAnOrder: jest.fn(() => {
      return () => Promise.resolve();
    }),
  };
});

describe(null, () => {
  it('render() with isPaid as true', () => {
    const isPaid = true;
    const id = 123;
    const store = mockStore({
      ui: uiStateMock,
      userOrders: userOrdersStateMock,
    });
    const wrapper = shallow(<Actions isPaid={isPaid} id={id} />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('render() with isPaid as false', () => {
    const isPaid = false;
    const id = 123;
    const store = mockStore({
      ui: uiStateMock,
      userOrders: userOrdersInitialState,
    });
    const wrapper = shallow(<Actions isPaid={isPaid} id={id} />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('Button.onClick()', () => {
    const isPaid = false;
    const id = 123;
    const store = mockStore({
      ui: uiStateMock,
      userOrders: userOrdersStateMock,
    });
    setupDispatch({ store });
    const wrapper = shallow(<Actions isPaid={isPaid} id={id} />, { context: { store } });
    const dummyWrapper = wrapper.dive();
    const buttonInstance = dummyWrapper
      .find(Button)
      .dive()
      .instance() as Button;
    buttonInstance.props.onClick(null);
    expect(payAnOrder).toBeCalledWith(id);
  });
  it('Button.loading', () => {
    const isPaid = false;
    const id = 123;
    const currentPaymentId = id;
    const store = mockStore({
      ui: uiStateMock,
      userOrders: { ...userOrdersStateMock, currentPaymentId },
    });
    setupDispatch({ store });
    const wrapper = shallow(<Actions isPaid={isPaid} id={id} />, { context: { store } });
    const dummyWrapper = wrapper.dive();
    const buttonInstance = dummyWrapper
      .find(Button)
      .dive()
      .instance() as Button;
    expect(buttonInstance.props.loading).toBeTruthy();
  });
});
