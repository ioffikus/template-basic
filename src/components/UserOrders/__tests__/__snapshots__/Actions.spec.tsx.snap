// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() with isPaid as false 1`] = `
<div
  className="root"
>
  <Button
    ghost={false}
    key="button"
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    size="small"
    type="primary"
  >
    Pay
  </Button>
</div>
`;

exports[`render() with isPaid as true 1`] = `
<div
  className="root"
/>
`;
